mutex/src/start-lock-workflow.ts

```diff
@@ -0,0 +1,21 @@
+import { Connection, Client } from '@temporalio/client';
+import { lockWorkflow } from './workflows';
+import { v4 as uuidv4 } from 'uuid';
+
+async function run() {
+  const connection = await Connection.connect();
+
+  const client = new Client({ connection });
+
+  const workflowId = 'lock-' + uuidv4();
```

Comment:
There's no need to explicitly start this workflow, in a real world example you'd use signalWithStart (from a local activity).
Please use the Go sample as reference: https://github.com/temporalio/samples-go/blob/e7f07468217806f57b22205692c4d70608b3e68b/mutex/mutex_workflow.go#L45

Reply:
@bergundy I wasn't able to figure out how to handle signalWithStart from an activity like the Go SDK examples do. It doesn't look like TS SDK's `Context` class lets you access the Client. And there doesn't seem to be any examples of that to work off of in the SDK tests or sample apps.

I'm thinking a cleaner approach would be to just start the lock Workflow from the same client script as the test Workflow like I did here: https://github.com/temporalio/samples-typescript/pull/222/commits/f8102f027823d45540dd0f86dcc141647f329cd3 . Are you ok with that or is there an alternative approach you prefer?

Reply:
Ah I see what Go does

https://github.com/temporalio/samples-go/blob/e7f07468217806f57b22205692c4d70608b3e68b/mutex/mutex_workflow.go#L138

We tell people to import from `@temporalio/client` and create a new Client. Seems nice to add it to Context though? @bergundy

Reply:
I think we'll end up doing that, Spencer and I have both mentioned it before.
It's useful.
In the meantime, you can inject a client via interceptors or though a class constructor.
Here's an interceptor we use for testing the SDK: https://github.com/temporalio/sdk-typescript/blob/main/packages/test/src/activities/interceptors.ts

Reply:
For this sample, to me interceptors feels like an unnecessary complication, and creating a Client and injecting it like below is simpler

https://github.com/temporalio/samples-typescript/blob/75cb3264bf28517a4aeff95edcbac147faba1c2d/activities-dependency-injection/src/worker.ts#L17

Reply:
I generally tend to agree.
If you want you can use a class instead of this factory pattern.